Notes for 25/7/2025 meeting (#185)
gammapy/gammapy-meetings: dev-meetings/2025/2025-07-25/README.md

AD: Dedicated call for changelog?
AS: When Régis comes back. Next week I'm not there.
AS: Am I missing important features we want to mention in the highlights?

- Emailadress to be used to send requests to, to sign up to gammapy? BK agrees, should be received by lead-devs. Can be done fast.

- QR agrees to lead meeting next time, as RT not back and AS can only stay for half an hour approximately.

### Open issues for v2.0

https://github.com/gammapy/gammapy/issues?q=is%3Aissue%20state%3Aopen%20milestone%3A2.0

- #5931: extern/xmltodict.py deprecated. BK,QR,AS agree to remove it completely, as code would be saved in old code-versions.
KF assigned.

- #5891: QR: We can postpone it, no usecase for which its really broken.

- #5854: AS: Would have been nice if we would have had time. BK, AS agree it would be nice, put it to 2.1 for now.

- #5783: AS: Not sure if this is a bug? Internally it works correct, annoying unit written. Agree: Not a bug, clean-up.

- #5775: BK: Needed for data challenge analysis, we have a bit of time. QR: Missing to link output of parser to select observation, one last thing was missing. I can finish it. QR assigned.

> - #5745: KF explains, can update docstring. AS: You could look at tutorial if it satisfies ??? KF agrees.

Reviewer: ```suggestion
- #5745: KF explains, can update docstring. AS: You could look at tutorial if it satisfies the PR. KF agrees, and can add links to the tutorial in the docstring.
```

> - #5737 "Adapt SensitivityEstimator to Cowan approach" : AS, QR discuss it. QR: We can do that, a flux point that does it ???. AS: Question of documentation. ??? AS: We need more discussion. BK: Add to userguide section for sensitivity computation. AS: Pointsource tutorial (CTA 1d)? Agree that flux point is not well exposed. BK: It is not 1D, can be 3D, we need upgrade in the documentation. Show what are statistical differences, what is the physical meaning. AS: Not a tutorial but userguide. BK: It is in the middle, I think we need a dedicated documentation here.

Reviewer: ```suggestion
- #5737 "Adapt SensitivityEstimator to Cowan approach" : AS, QR discuss it. QR: We can do that, a flux point that does it AS: Question of documentation.  AS: We need more discussion. BK: Add to userguide section for sensitivity computation. AS: Pointsource tutorial (CTA 1d)? Agree that flux point is not well exposed. BK: It is not 1D, can be 3D, we need upgrade in the documentation. Show what are statistical differences, what is the physical meaning. AS: Not a tutorial but userguide. BK: It is in the middle, I think we need a dedicated documentation here.
```

AS: Put it to 2.0 if someone has time to put documentation in, no one assigned.

> QK: We could make new class Joint Sensitivity/Fluxpoint (???) Estimator?

Reviewer: ```suggestion
QK: We could make new class Joint Sensitivity Estimator? That would combine the steps to do this. Currently it is tied with points estimators.
```

AS: Question of documentation.

> - #5728: AS: More or less necessary. QR: PR for 2.1 ???

Reviewer: ```suggestion
- #5728: AS: More or less necessary. QR: PR moved to 2.1, need time for proper reviews.
```

- #5713 "Adapt reoptimize option for FluxPointEstimator":  AS,QR discuss. Expose in doc that user can config. parameters to fit before running estimator. AS comments in Issue.

- #5705 assigned to LS, put to 2.1.

- #5693 "Request to Add Finke et al. (2022) EBL Model.": 
AS suggests: Solution how to write your own model, that can be read by gammapy.
QR agrees, would be better if we do not need to add models for other people.

- #5691: AS presents.

- #5626: AS; BK agree, Not needed, can be moved to 2.1

- #5598: BK: 2.1, still a lot of work to do.

> - #5587: "Enhance CI": FK: Can be moved to 2.1. You can check preview docs in PR and use labels to skip jobs. Main things remain if we want to rebuild CI completely. ??? DM assigned, for 2.1.

Reviewer: ```suggestion
- #5587: "Enhance CI": KF: Can be moved to 2.1. You can check preview docs in PR and use labels to skip jobs. Main things remain if we want to rebuild CI completely, which would involved something like CircleCI. DM assigned, for 2.1.
```

- #5552: BK: 2.1.

- #5494: AS: What is state? No failing tests? QR: 2.1, we can simplify tests maybe.

> - #5484: BK: Issue that it is super slow, it works. QK: Testdata was not good? BK: We have to check to have proper evaluation.
> QR and BK discuss. AS: Cant we update our HAWC-tutorial? QR: Not really the same. FK: Not really a place in the tut. for this. BK: Thereby adding tutorial. AS: No public data for this. QR: I check in the tutorial. Internally histogram is fitted and sometimes fit is terrible.

Reviewer: ```suggestion
- #5484: BK: Issue that it is super slow, it works. QR: Test data was not good? BK: We have to check to have proper evaluation.
QR and BK discuss. AS: Cant we update our HAWC-tutorial? QR: Not really the same. KF: Not really a place in the tutorial for this. BK: Thereby adding tutorial. AS: No public data for this. QR: I check in the tutorial. Internally histogram is fitted and sometimes fit is terrible.
```

BK: Do we have benchmark on SWGO? QR: Not in gammapy. BK: Could be interesting in gammapy. BK: 2.1

- #4712: AS: Set to 2.1

- #4641: AS: Claudio planned this. Set to 2.1

AS: Most other Issues are older than 2 years, are there important ones staying in 2.0 or move all to 2.1?
QR: Yes, you can move all of them.
KF: Maybe some repeat? Maybe we should go through and check? I can have a quick look and let you know.

### Open PRs for v2.0

https://github.com/gammapy/gammapy/pulls?q=is%3Aopen+is%3Apr+milestone%3A2.0

* #6014 : QR: We should support it, AS: Ok from my side, if someone else reviews we can merge.
 
* #6013 : DM: explains AS: Should be fine, can do review.

* #5838 : AS: Put to 2.1 QR agrees.

* #5859 : AS: Will take a look why test is failing.

### PRs already discussed at least once:

* [5982] "Add class instance caching for WcsGeom":
  
QR: Would be good to have in for 2.0, is deactivated by default, so will not cause problems.
AS: AD, you suggested to do benchmarks, where it helps, can you do review on this?
AD: Yes, sure!

> BK asks about details, suggests to make a reminder for it ???

Reviewer: ```suggestion
BK asks about details, suggests to make a reminder for it 
```

* [5965] Non-detected sources

AS: Long open, basically ready, only if licence should go on top.
BK: Will tell you this afternoon by slack if any comments.
AS: Otherwise could go in.
KF: Should I check again, some comments from my first review not implemented. KF agrees to implement them.

* [5954] "Separate internal Eventlist data model from gadf format"
  
QR: Check test? AS checks, was okay. BK: Some more testing needed.
AS: We wait for Régis until he is back.

* [5949] Adapt tutorial order
  
KF: Open to suggestions for minimal solution that we could implement. now.
AS: Idea instead of calling it API, Keysteps.
QR: Observationclustering and DM maybe into other section?
AS: Other aspects (nested sampling) also not key to gammapy analysis.
KF: Suggestion from first entry in Issue #5796, BK agrees. KF: I'm open to suggestions, but removing API is good idea, it could be seen and neglected as not important by people.
AS: "Mask" as advanced options?

KF: Maybe really splitting it, do it as suggested in bottom of first entry of Issue #5796.

> QR: EBL, Pulsars, DM, Energy-dep morph. into astrophysics section. Nested sampling should not go in, not astroph. use case (QR), as it is fitting ??? (BK).

Reviewer: ```suggestion
QR: EBL, Pulsars, DM, Energy-dep morph. into astrophysics section. Nested sampling should not go in, not astroph. use case (QR), as it is fitting and already exposed in the user guide under fitting (BK).
```

AS: Second section "Detailed explanations" (How to use estimators, catalogs,...)
BK: agrees.

* [5910] "Change argument name in ul"
  
KF: Docstring did not match implementation. In the end implemented Régis suggestion, AS helped me. But one test still failing. Added more raised errors.
AS explains. Get minima from interpolated profile.
KF: Unclear why test fails, as only in one case.
AS: Different interplation to be more stable, but consistent to use different ones? 
LS: Can type of polynomials have an effect, probably preventing finding minimum properly by artifacts?

> QR anwers: Yes, but here shape of interpolant similar to expected shape.

Reviewer: ```suggestion
QR answers: Yes, but here shape of interpolant similar to expected shape.
```

### PRs opened last week (less than 8 days ago): 
* [#6010](https://github.com/gammapy/gammapy/pull/6010) Changelog for 2.0 - Atreyee Sinha
  
- AS: Started to fill in Changelog. Am not aware of the things to be highlighted for Infrastructure: DM and KF agree to take a look.
- AS: Get a timeout, KF agrees to check it, AD: Is a limitation by GitHub Actions. Limit by GitHub for too many requests in a too small amount of time, you could artificially increase time. "time.sleep" in between requests could work.

* [#6009](https://github.com/gammapy/gammapy/pull/6009) Update dev docs to mention utility scripts available for a release - Atreyee Sinha

- AS: Where does metadata reside in workflow? BK, AD answer. AD recommends to run it, see what diff is and if it is reasonable.

> - AS: What does file prepare-release.py? AD: Updates release-date and version in citation.cff. Suggests to run it and see diff. AS: Before or after release? AD: Before the release. BK: After or before release-candidate is made? AD: Not sure it matters. As release candidate does not go to zenodo but only PyPI. ???

Reviewer: ```suggestion
- AS: What does file prepare-release.py? AD: Updates release-date and version in citation.cff. Suggests to run it and see diff. AS: Before or after release? AD: Before the release. BK: After or before release-candidate is made? AD: Not sure it matters. As release candidate does not go to zenodo but only PyPI. 
```

### issues opened last week (less than 8 days ago): 
* [#6003](https://github.com/gammapy/gammapy/issues/6003) FluxPoints documentation showing extra attribute - None
  

* [#5990](https://github.com/gammapy/gammapy/issues/5990) Prior class - weight parameter in the call, not the evaluate - None

AS, QR: is about, how should the weights go in. For 2.0 we can keep it like this, but we need to clarify purpose of weight. Two different cases, QR explains. AD: From stat. perspective, it corresponds to variance (at least for Gaussian case). For Gaussian prior already in, as you define variance of prior. Question if you want to support this in more uniform way. For uniform prior also in, once you define bounds of uniform prior. Some degree of overlap between definitions, matter of documentation? 
QR: For now, only defined as multiplicative factor.

> AD: Documentation, have one example where we show how to do this. Should be possible to handle with definition of variance ???

Reviewer: ```suggestion
AD: Documentation, have one example where we show how to do this. Should be possible to handle with definition of variance
```
